lib/internal/loader/Loader.js

```diff
@@ -70,6 +79,14 @@ class Loader {
     const module = await job.run();
     return getNamespaceOfModuleWrap(module);
   }
+
+  static registerImportDynamicallyCallback(loader) {
+    setImportModuleDynamicallyCallback(async (referrer, specifier) => {
+      const normalizedBase = normalizeReferrerURL(referrer);
```

Comment:
I take it this `referrer` path context is provided directly from V8 for scripts? What does it become for `eval` and `vm.runInThisContext` cases?

Reply:
Yep, it's 1:1 the `ScriptOrigin#url` as provided by V8. For `vm.runInThisContext` it's "evalmachine.anonymous" by default but can be overridden using the `filename` option.

For both direct and indirect eval `referrer` is null - nice find! Will add test cases and make sure they fail cleanly.

Reply:
Updated with test cases that document the behavior for `eval`: https://github.com/nodejs/node/pull/15713/commits/3eaf5fe46e666eed6febba7a69884d5a7ec1a7fa

Reply:
Is this the final stance v8 is taking or will this API be changing at all do you know? If it is pretty much set, then I guess we should just adjust the resolver to handle undefined `referrer` as defaulting to the standard base referrer.

Reply:
@guybedford this is a bug, V8 6.1 implements `import()` but 6.2 is the recommended implementations. See https://tc39.github.io/ecma262/#sec-getactivescriptormodule , which will be given instead of a string. So the C++ API changes a bit as well.

Reply:
@bmeck that seems to be more sensible, thanks for the link.